From a7a42a3e6ba189ff773beeac106d956327f747c8 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 19 Jul 2024 10:23:38 +0200
Subject: [PATCH 1178/1695] dts: stm32: LTDC secure/non-secure pinctrl for
 stm32mp13f-dk

Define LTDC secure and non-secure pinctrl configurations for
stm32mp13f-dk board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8f6f06a13ac854bbb6250f7987f73d9080aa6922 dts: stm32: LTDC secure/non-secure pinctrl for stm32mp13f-dk]
Change-Id: I1823d78f086a7d1cf2ec0bcb999a69440088f83c
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/394348
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/dts/stm32mp135f-dk.dts | 22 ++++++++++++++++++++--
 1 file changed, 20 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index f6e93de42..65db7cf62 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -386,8 +386,26 @@
 
 &ltdc {
 	pinctrl-0 = <&ltdc_pins_a>;
-	access-controllers-conf-nsec = <&etzpc DECPROT(STM32MP1_ETZPC_LTDC_ID, DECPROT_NS_RW, DECPROT_UNLOCK)>;
-	access-controllers-conf-sec = <&etzpc DECPROT(STM32MP1_ETZPC_LTDC_ID, DECPROT_S_RW, DECPROT_UNLOCK)>;
+	access-controllers-conf-nsec =
+		<&etzpc DECPROT(STM32MP1_ETZPC_LTDC_ID, DECPROT_NS_RW, DECPROT_UNLOCK)>,
+		<&gpioa (GPIO_STM32_NSEC | TZPROT(15))>,
+		<&gpiob (GPIO_STM32_NSEC | TZPROT(6) | TZPROT(12))>,
+		<&gpioc (GPIO_STM32_NSEC | TZPROT(6) | TZPROT(7))>,
+		<&gpiod (GPIO_STM32_NSEC | TZPROT(5) | TZPROT(9) | TZPROT(10) | TZPROT(14))>,
+		<&gpioe (GPIO_STM32_NSEC | TZPROT(0) | TZPROT(7) | TZPROT(9) | TZPROT(13))>,
+		<&gpiof (GPIO_STM32_NSEC | TZPROT(1) | TZPROT(2) | TZPROT(3))>,
+		<&gpiog (GPIO_STM32_NSEC | TZPROT(0) | TZPROT(4) | TZPROT(7))>,
+		<&gpioh (GPIO_STM32_NSEC | TZPROT(9) | TZPROT(13) | TZPROT(14))>;
+	access-controllers-conf-sec =
+		<&etzpc DECPROT(STM32MP1_ETZPC_LTDC_ID, DECPROT_NS_R_S_W, DECPROT_UNLOCK)>,
+		<&gpioa (TZPROT(15))>,
+		<&gpiob (TZPROT(6) | TZPROT(12))>,
+		<&gpioc (TZPROT(6) | TZPROT(7))>,
+		<&gpiod (TZPROT(5) | TZPROT(9) | TZPROT(10) | TZPROT(14))>,
+		<&gpioe (TZPROT(0) | TZPROT(7) | TZPROT(9) | TZPROT(13))>,
+		<&gpiof (TZPROT(1) | TZPROT(2) | TZPROT(3))>,
+		<&gpiog (TZPROT(0) | TZPROT(4) | TZPROT(7))>,
+		<&gpioh (TZPROT(9) | TZPROT(13) | TZPROT(14))>;
 	status = "okay";
 };
 
-- 
2.47.2

